apps/web/pages/_document.tsx

```diff
@@ -67,7 +67,39 @@ class MyDocument extends Document<Props> {
             nonce={nonce}
             id="newLocale"
             dangerouslySetInnerHTML={{
-              __html: `window.calNewLocale = "${newLocale}";`,
+              __html: `
+              window.calNewLocale = "${newLocale}";
+              (function applyTheme() {
+                try {
+                  const appTheme = localStorage.getItem('app-theme');
+                  if (!appTheme) return;
+
+                  let bookingTheme, username;
+                  for (let i = 0; i < localStorage.length; i++) {
+                    const key = localStorage.key(i);
+                    if (key.startsWith('booking-theme:')) {
+                      bookingTheme = localStorage.getItem(key);
+                      username = key.split("booking-theme:")[1];
+                      break;
+                    }
+                  }
+
+                  const onReady = () => {
+                    const isBookingPage = username && window.location.pathname.slice(1).startsWith(username);
+
+                    if (document.body) {
+                      document.body.classList.add(isBookingPage ? bookingTheme : appTheme);
+                    } else {
+                      requestAnimationFrame(onReady);
+                    }
+                  };
+
+                  requestAnimationFrame(onReady);
+                } catch (e) {
+                  console.error('Error applying theme:', e);
+                }
+              })();
```

Comment:
Could we add some unit tests for this. We could make it a function and use fn.String() here.


This logic works in combination with app-providers-app-dir.tsx. So, if possible we should share constants b/w them e.g. 'booking-theme' prefix.


Also, I am not clear why we need to add this logic here. Doesn't AppProviders stuff work now?

Reply:
No; at the moment there's a brief moment between pageload and app render where it decides on the wrong theme (only when dark mode is selected)